Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSMachHeaders.m

```diff
@@ -88,120 +68,58 @@ BOOL IsUnfairLockSupported(NSProcessInfo *processInfo) {
 #elif BSG_PLATFORM_WATCHOS
     minSdk.majorVersion = 3;
 #endif
-    return [processInfo isOperatingSystemAtLeastVersion:minSdk];
-}
-
-void bsg_check_unfair_lock_support() {
-    bsg_unfair_lock_supported = IsUnfairLockSupported([NSProcessInfo processInfo]);
+    return [[NSProcessInfo processInfo] isOperatingSystemAtLeastVersion:minSdk];
 }
 
-// MARK: - Replicate the DYLD API
-
-uint32_t bsg_dyld_image_count(void) {
-    return bsg_mach_binary_images.used;
-}
+static void (*bsg_mach_headers_cache_lock_func)(void) = NULL;
+static void (*bsg_mach_headers_cache_unlock_func)(void) = NULL;
 
-const struct mach_header* bsg_dyld_get_image_header(uint32_t imageIndex) {
-    if (imageIndex < bsg_mach_binary_images.used) {
-        return bsg_mach_binary_images.contents[imageIndex].header;
+void bsg_mach_headers_cache_lock() {
```

Comment:
The docs for `[NSProcessInfo processInfo]` states the return value is cached after the first invocation. Therefore I think the null check and assignment to static variable can be removed here as it will be clearer if `bsg_unfair_lock` and `bsg_spin_lock` are just invoked directly

Reply:
That would certainly be cleaner! It seems that is an efficient call, so I'll make the change.